apps/cms-v2/src/modules/providers/payloadcms/payloadcms-config-form.tsx

```diff
@@ -0,0 +1,263 @@
+ * todo react on token change, refresh mutation
+ */
+const PureForm = ({ defaultValues, onSubmit, onDelete }: PureFormProps) => {
+  const { notifyError } = useDashboardNotification();
+
+  const { control, getValues, setValue, watch, handleSubmit, clearErrors, setError } = useForm({
+    defaultValues: defaultValues,
+    resolver: zodResolver(PayloadCmsProviderConfig.Schema.Input.omit({ type: true })),
+  });
+
+  return (
+    <Box
+      as="form"
+      display={"grid"}
+      gap={4}
+      onSubmit={handleSubmit((vals) => {
+        onSubmit(vals);
+      })}
+    >
+      <Input
+        required
+        control={control}
+        name="configName"
+        label="Configuration name"
+        helperText="Meaningful name that will help you understand it later. E.g. 'staging' or 'prod' "
+      />
+      <Box display={"grid"} gap={4} marginY={4}>
+        <Text variant="heading">Provide connection details</Text>
+        <Input
+          required
+          control={control}
+          name="payloadApiUrl"
+          type="url"
+          label="API url"
+          helperText="URL where Payload API is available. By default ends with /api"
+        />
+
+        <Box
+          backgroundColor="surfaceNeutralHighlight"
+          borderColor="neutralHighlight"
+          borderWidth={1}
+          borderStyle={"solid"}
+          padding={4}
+          borderRadius={4}
+        >
+          <Text variant="heading" as="h1" marginBottom={4}>
+            Authorization
+          </Text>
+          <Text marginBottom={2} as="p">
+            Payload can be configured to have open operations (not recommended) or to require an API
+            key. Key can be generated per user. To authenticate, you need to provide both user slug
+            and the key itself.{" "}
+          </Text>
+          <TextLink
+            marginBottom={2}
+            display="block"
+            newTab
+            href="https://payloadcms.com/docs/authentication/config"
+          >
+            Read more in Payload docs
+          </TextLink>
+          <Text as="p" marginBottom={4}>
+            If your API is open (e.g. for development purposes) leave both fields empty.
+          </Text>
+
+          <Input
+            control={control}
+            name="authenticatedUserSlug"
+            label="Authenticated user slug"
+            placeholder="e.g. apps"
+          />
+          <Input control={control} name="authToken" type="password" label="User API Key" />
+        </Box>
+      </Box>
+      <Box display={"grid"} gap={4} marginY={4}>
+        <Text variant="heading">Configure fields mapping</Text>
+        <Input
+          label="Collection Slug"
+          name="collectionName"
+          control={control}
+          helperText="Slug of your collection in Payload, e.g. 'saleorVariants'"
+        />
+
+        <React.Fragment>
+          <Text as="p" variant="heading" size="small">
+            Map fields from Saleor to your contentful schema.
```

Comment:
**Typo[^1]**:
```suggestion
            Map fields from Saleor to your content schema.
```
[^1]: 🛑 Should be blocking the PR

Reply:
Or `Payload CMS` schema, not sure which is better in this context

Reply:
yea it was Contentful and I just copypasted, will revisit